### PR TITLE
closes [make disclaimer optional #34]

### PR DESCRIPTION
**Pull request by matthewcornell**
Changes:

**src/:**
- predtimechart.js: passing isDislaimerPresent to _createUIElements() and initializeUI() so they can not create a <P> if not present (former), and not set it (latter)
- schema.json: made `disclaimer` property not required
- schema-validator.cjs: generated from schema.json by `npm run ajv_compile` command

**test/:**
- ui.js: adds a test to drive coding where elements with the `forecastViz_disclaimer` class should not be present if no disclaimer, and vice versa
- validation.js: adds a case to ensure that disclaimer is optional in validation (drives changing schema.json)

### CODE REVIEW

File: src/predtimechart.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ function _setSelectedTruths() {
  * @param taskIdsKeys - array of options.task_ids keys. used to create task rows - one per task ID
  * @private
  */
-function _createUIElements($componentDiv, isUemEnabled, taskIdsKeys) {
+function _createUIElements($componentDiv, isUemEnabled, taskIdsKeys, isDislaimerPresent) {
```
Comment (zkamvar) [inline src/predtimechart.js:107]:
```suggestion
function _createUIElements($componentDiv, isUemEnabled, taskIdsKeys, isDisclaimerPresent) {
```

File: src/predtimechart.js [hunk 2]
```diff
@@ -195,7 +195,9 @@ function _createUIElements($componentDiv, isUemEnabled, taskIdsKeys) {
         '    </div>\n' +
         '</div>'
     );
-    $vizDiv.append($('<p class="forecastViz_disclaimer"><b><span id="disclaimer">(disclaimer here)</span></b></p>'));
+    if (isDislaimerPresent) {
```
Comment (zkamvar) [inline src/predtimechart.js:198]:
```suggestion
    if (isDisclaimerPresent) {
```

File: src/predtimechart.js [hunk 3]
```diff
@@ -429,8 +431,9 @@ const App = {
 
         console.debug('initialize(): initializing UI');
         const $componentDiv = $(componentDivEle);
-        _createUIElements($componentDiv, this.isUemEnabled, Object.keys(this.state.task_ids));
-        this.initializeUI(options['initial_task_ids']);
+        const isDislaimerPresent = options.hasOwnProperty('disclaimer');
+        _createUIElements($componentDiv, this.isUemEnabled, Object.keys(this.state.task_ids), isDislaimerPresent);
+        this.initializeUI(options['initial_task_ids'], isDislaimerPresent);
```
Comment (zkamvar) [inline src/predtimechart.js:436]:
```suggestion
        const isDisclaimerPresent = options.hasOwnProperty('disclaimer');
        _createUIElements($componentDiv, this.isUemEnabled, Object.keys(this.state.task_ids), isDisclaimerPresent);
        this.initializeUI(options['initial_task_ids'], isDisclaimerPresent);
```

File: src/predtimechart.js [hunk 4]
```diff
@@ -527,7 +530,7 @@ const App = {
             window.history.replaceState(newUrl.toString(), '', newUrl);
         }
     },
-    initializeUI(initial_task_ids) {
+    initializeUI(initial_task_ids, isDislaimerPresent) {
```
Comment (zkamvar) [inline src/predtimechart.js:533]:
```suggestion
    initializeUI(initial_task_ids, isDisclaimerPresent) {
```

File: src/predtimechart.js [hunk 5]
```diff
@@ -539,7 +542,9 @@ const App = {
         this.updateTruthAsOfCheckboxText();
 
         // initialize disclaimer
-        $('#disclaimer').text(this.state.disclaimer);
+        if (isDislaimerPresent) {
```
Comment (zkamvar) [inline src/predtimechart.js:545]:
```suggestion
        if (isDisclaimerPresent) {
```

**Review by zkamvar — CHANGES_REQUESTED** [review]
This looks okay to me. I am not familiar with the workflow for testing locally, but this logic looks sound. The only thing that should be fixed is the misspelling of "Disclaimer" in `isDislaimerPresent`. 

If I understand correctly, this will allow https://github.com/hubverse-org/hub-dashboard-predtimechart to use null disclaimer values?

**Review by zkamvar — DISMISSED** [review]

**Comment by matthewcornell** [comment]
@matthewcornell re: "... this will allow https://github.com/hubverse-org/hub-dashboard-predtimechart to use null disclaimer values?" Actually, it will allow omitting `disclaimer` (properly spelled :-). I thought that expressed "no disclaimer" better than a required property with special values.

**Review by elray1 — APPROVED** [review]

**Pull request merged**